metrics/release_test.go

```diff
@@ -17,239 +17,225 @@ limitations under the License.
-				startTime := metav1.NewTime(creationTime.Add(time.Second * time.Duration(seconds)))
-				elapsedSeconds += seconds
-				RegisterNewRelease(creationTime, &startTime)
-			}
-			Expect(testutil.ToFloat64(ReleaseConcurrentTotal)).To(Equal(float64(len(inputSeconds))))
+		It("increments ReleaseTotal", func() {
+			Expect(testutil.ToFloat64(ReleaseTotal.WithLabelValues("", "", "", "", "", "", ""))).To(Equal(float64(0)))
+			RegisterCompletedRelease(startTime, completionTime, "", "", "", "", "", "", "")
+			Expect(testutil.ToFloat64(ReleaseTotal.WithLabelValues("", "", "", "", "", "", ""))).To(Equal(float64(1)))
 		})
 
-		It("registers a new observation for 'release_attempt_running_seconds' with the elapsed time from the moment"+
-			"the Release was created to when it started (Release marked as 'Running').", func() {
-			// Defined buckets for ReleaseAttemptRunningSeconds
-			timeBuckets := []string{"1", "5", "10", "30"}
-			data := []int{1, 2, 3, 4}
-			readerData := createHistogramReader(AttemptRunningSecondsHeader, timeBuckets, data, "", elapsedSeconds, len(inputSeconds))
-			Expect(testutil.CollectAndCompare(ReleaseAttemptRunningSeconds, strings.NewReader(readerData))).To(Succeed())
+		It("increments ReleaseTotal", func() {
+			RegisterCompletedRelease(startTime, completionTime,
+				releaseTotalLabels[0],
+				releaseTotalLabels[1],
+				releaseTotalLabels[2],
+				releaseTotalLabels[3],
+				releaseTotalLabels[4],
+				releaseTotalLabels[5],
+				releaseTotalLabels[6],
+			)
+			Expect(testutil.CollectAndCompare(ReleaseTotal,
+				newCounterReader(
+					releaseTotalOpts,
+					releaseTotalLabels,
+				))).To(Succeed())
 		})
 	})
 
-	Context("When RegisterCompletedRelease is called", func() {
-		BeforeAll(func() {
-			ReleaseAttemptRunningSeconds = prometheus.NewHistogram(
-				prometheus.HistogramOpts{
-					Name:    "release_attempt_running_seconds",
-					Help:    "Release durations from the moment the release resource was created til the release is marked as running",
-					Buckets: []float64{1, 5, 10, 30},
-				},
-			)
-			ReleaseDurationSeconds = prometheus.NewHistogramVec(
-				prometheus.HistogramOpts{
-					Name:    "release_attempt_duration_seconds",
-					Help:    "Release durations from the moment the release PipelineRun was created til the release is marked as finished",
-					Buckets: []float64{60, 600, 1800, 3600},
-				},
-				[]string{"reason", "strategy", "succeeded", "target"},
-			)
+	Context("When RegisterCompletedReleaseDeployment is called", func() {
+		var completionTime, startTime *metav1.Time
 
-			ReleaseConcurrentTotal = prometheus.NewGauge(
-				prometheus.GaugeOpts{
-					Name: "release_attempt_concurrent_requests",
-					Help: "Total number of concurrent release attempts",
-				},
-			)
-			metrics.Registry.MustRegister(ReleaseAttemptRunningSeconds, ReleaseDurationSeconds, ReleaseConcurrentTotal)
+		BeforeEach(func() {
+			initializeMetrics()
+
+			completionTime = &metav1.Time{}
```

Comment:
there should be a test about doing nothing when the completion or start time are nil

Reply:
Added.